app/script/localization/Localizer.js

```diff
@@ -48,51 +48,42 @@ class Localizer {
 z.localization.Localizer = new Localizer();
 
 z.l10n = (() => {
-  const replaceTags = (string, tagSubstitutes) => {
-    Object.entries(tagSubstitutes).forEach(([identifier, substitute]) => {
-      string = string.replace(new RegExp(`\\[${identifier}\\]`, 'g'), substitute);
-    });
+  const isStringOrNumber = toTest => _.isString(toTest) || _.isNumber(toTest);
 
-    return string;
-  };
-
-  const replaceWithArray = (string, substitutions) => {
-    substitutions.forEach(([identifier, substitute]) => {
-      string = string.replace(new RegExp(`{{${identifier}}}`, 'g'), substitute);
-    });
-
-    return string;
+  const replaceSubstitute = (string, regex, substitute) => {
+    if (isStringOrNumber(substitute)) {
+      return string.replace(regex, substitute);
+    }
+    return string.replace(
+      regex,
+      (found, content) => (substitute.hasOwnProperty(content) ? substitute[content] : found)
```

Comment:
Can we move this to a const `replaceFunction` so that the actual replace call turns out to be a oneliner?

Reply:
prettier strikes again ...

Reply:
Or maybe even:
````
const replacement = isStringOrNumber(substitute)
  ? substitute
  : (found, content) => (substitute.hasOwnProperty(content) ? substitute[content] : found);+      
return string.replace(regex, replacement);
````